### PR TITLE
vm support: allow multiple Istio Proxies on a single VM 

### PR DESCRIPTION
**Pull request by yskopets**
**Please provide a description of this PR:**

This PR implements the design proposal from https://github.com/istio/istio/issues/34572.

At the moment, there are no new unit/integration tests in this PR to let reviewers to focus on actual changes.

### CODE REVIEW

**Comment by marcbachmann** [comment]
> This PR implements the design proposal from https://github.com/istio/istio/pull/37876.

this links to this PR itself 😉

**Comment by yskopets** [comment]
@marcbachmann Yeah, sorry. I've updated the link.

File: pilot/cmd/pilot-agent/status/util/stats.go [hunk 1]
```diff
@@ -73,7 +74,7 @@ func GetReadinessStats(localHostAddr string, adminPort uint16) (*uint64, bool, e
 		localHostAddr = "localhost"
 	}
 
-	readinessURL := fmt.Sprintf("http://%s:%d/stats?usedonly&filter=%s", localHostAddr, adminPort, readyStatsRegex)
+	readinessURL := fmt.Sprintf("http://%s/stats?usedonly&filter=%s", net.JoinHostPort(localHostAddr, strconv.Itoa(int(adminPort))), readyStatsRegex)
```
Comment (adiprerepa) [inline pilot/cmd/pilot-agent/status/util/stats.go:77]: Why this change?

File: pilot/cmd/pilot-agent/status/server.go [hunk 2]
```diff
@@ -181,12 +185,17 @@ func NewServer(config Options) (*Server, error) {
 
 	probes = append(probes, config.Probes...)
 	s := &Server{
+		address:               "", // effectively, 0.0.0.0
```
Comment (adiprerepa) [inline pilot/cmd/pilot-agent/status/server.go:188]: why not set to `0.0.0.0` direct?
